cvat-canvas/src/typescript/interactionHandler.ts

```diff
@@ -411,15 +424,100 @@ export class InteractionHandlerImpl implements InteractionHandler {
                 }
             }
         });
+        window.addEventListener('keyup', (e: KeyboardEvent): void => {
+            if (!this.checkCursorPosition()) {
+                if (e.keyCode === 17) {
+                    const store = getCVATStore();
+                    const state = store.getState();
+                    const {
+                        settings: {
+                            workspace: {
+                                blockMode,
+                            },
+                        },
+                        annotation: {
+                            canvas: {
+                                activeControl,
+                            },
+                        },
+                    } = state;
+                    if ([ActiveControl.AI_TOOLS].includes(activeControl)) {
+                        // on interactors
+                        this.interactBlockMode(false, false);
+                    } else if ([ActiveControl.OPENCV_TOOLS].includes(activeControl) && !this.thresholdWasModified) {
+                        // on scissors
+                        this.thresholdWasModified = false;
+                        this.interactBlockMode(!blockMode.enabled, true);
+                    }
+                }
```

Comment:
All this logic should be implemented in cvat-ui. 

Button state should be maintained, for example, using an additional callback parameter in ``interact`` method, to notice UI if block was enabled disabled: 
```
interact({
    onChangeBlockState?: ((blocked) => dispatch(switchBlockMode(blocked))); ///
    ....
});
```

Button visibility should be maintaned from ``tools-control.tsx``, ``opencv-control.tsx``.
I would recommend to update switchBlockMode signature to:
switchBlockMode({ buttonVisible?: boolean; enabled?: boolean })

where both parameters are optional and call is more implicit
```
switchBlockMode({ buttonVisible: true })
switchBlockMode({ enabled: true })
switchBlockMode({ buttonVisible: true, enabled: false })
```

Personally, I also would rename ``blockMode`` to something else, I do not think that ``blockMode``is informative enough. At least when I see it in the code, I do not understand what is it responsible for.

Maybe: ``toolsBlockerState: { buttonVisible: boolean; algorithmsLocked: boolean }``

Reply:
All this logic is removed to cvat-ui.